lib/ansible/template/__init__.py

```diff
@@ -589,7 +612,9 @@ def __init__(self, loader, shared_loader_obj=None, variables=None):
         self._fail_on_filter_errors = True
         self._fail_on_undefined_errors = C.DEFAULT_UNDEFINED_VAR_BEHAVIOR
 
-        self.environment = AnsibleEnvironment(
+        environment_class = AnsibleNativeEnvironment if USE_JINJA2_NATIVE else AnsibleEnvironment
```

Comment:
I question whether this should be an arg for `Templar`.  Just a thought, not a requirement.

Reply:
I have gone through like 20 iterations of implementing this feature and in like half of them I had `environment_class` as an argument for `Templar` and for a reason I can't remember anymore I decided this was easier.